src/experimental/tools/mediaCapabilitiesProber/probers/mediaContentTypeWithFeatures/index.ts

```diff
@@ -48,35 +51,29 @@ function isTypeSupportedWithFeaturesAPIAvailable(): Promise<void> {
  */
 export default function probeTypeWithFeatures(
   config: IMediaConfiguration
-) : Promise<[number]> {
+) : Promise<[ProberStatus]> {
   return isTypeSupportedWithFeaturesAPIAvailable().then(() => {
     const keySystem = config.keySystem;
     const type = keySystem ? (keySystem.type || "org.w3.clearkey") : "org.w3.clearkey";
-    const hdcp = config.hdcp;
-
-    const video = config.video;
-    const audio = config.audio;
-    const display = config.display;
-
-    const features = formatConfig(video, hdcp, audio, display);
+    const features = formatConfig(config);
```

Comment:
I thought it was more logical to give the whole config to a function called `formatConfig`.

Also it removes the need to define those intermediary interfaces